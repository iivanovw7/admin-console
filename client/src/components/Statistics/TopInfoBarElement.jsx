import { Avatar, Icon, ListItem, ListItemText } from '@material-ui/core';
import React from 'react';

export const TopInfoBarElement = props => {

  const { classes, icon, view, title } = props;

  return (
    <ListItem className={classes.desktop}>
      <Avatar>
        <Icon>{icon}</Icon>
      </Avatar>
      <ListItemText primary={title} secondary={view}/>
    </ListItem>
  );
};



