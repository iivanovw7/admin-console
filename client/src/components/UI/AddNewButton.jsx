import React from 'react';
import PropTypes from 'prop-types';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Button from '@material-ui/core/Button';
import withWidth from '@material-ui/core/withWidth';

const AddNewButton = props => {

  const { classes, width, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  const mobileButton = () => {
    return (
      <Button size="small" className={classes.margin}
              onClick={() => {
                handleClick();
              }}>
        NEW
      </Button>
    );
  };

  const desktopButton = () => {
    return (
      <Fab color="primary" size="small" aria-label="Add" className={classes.fab}
           onClick={() => {
             handleClick();
           }}>
        <AddIcon/>
      </Fab>
    );
  };

  const components = {
    xs: mobileButton(),
    sm: mobileButton()
  };

  const Component = components[width] || desktopButton();

  return (
    <div className={classes.formRoot}>
      {Component}
    </div>
  );

};

AddNewButton.propTypes = {
  width: PropTypes.string.isRequired
};

export default withWidth()(AddNewButton);

