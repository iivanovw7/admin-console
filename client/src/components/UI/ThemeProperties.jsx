import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

