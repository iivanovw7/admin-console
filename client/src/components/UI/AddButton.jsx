import { withStyles } from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Button from '@material-ui/core/Button';
import { Buttons } from './ThemeProperties';

const AddButton = props => {

  const { classes, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  return (
    <div className={classes.formRoot}>
      <Button size="small" className={classes.mobile}
              onClick={() => {
                handleClick();
              }}>
        <strong>Create</strong>
      </Button>
      <Fab color="primary" size="small" aria-label="Add" className={classes.desktop}
           onClick={() => {
             handleClick();
           }}>
        <AddIcon/>
      </Fab>
    </div>
  );

};

AddButton.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(Buttons, { withTheme: true })(AddButton);