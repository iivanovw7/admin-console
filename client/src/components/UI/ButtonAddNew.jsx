import { Button, Fab } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import AddIcon from '@material-ui/icons/Add';
import PropTypes from 'prop-types';
import React from 'react';
import { Buttons } from './ThemeProperties';

const ButtonAddNew = props => {

  const { classes, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  return (
    <div className={classes.formRoot}>
      <Button
        size="small"
        className={classes.mobile}
        onClick={() => {
          handleClick();
        }}
      >
        <strong>Create</strong>
      </Button>
      <Fab
        color="primary"
        size="small"
        aria-label="Add"
        className={classes.desktop}
        onClick={() => {
          handleClick();
        }}>
        <AddIcon/>
      </Fab>
    </div>
  );

};

ButtonAddNew.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(Buttons, { withTheme: true })(ButtonAddNew);