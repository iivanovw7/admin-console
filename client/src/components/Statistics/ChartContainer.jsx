import { Button, Grid, Paper } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Typography from '../../screens/Statistics';
import { Container } from '../UI/ThemeProperties';
import { sendMessage } from '../../actions';

const ChartContainer = props => {

  const { classes, history, dispatch, statsData } = props;

  return (
    <Grid item xs={12} sm={6} id={statsData}>
      <Paper className={classes.paper}>
        <h3>{statsData}</h3>
      </Paper>
      <Typography paragraph>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
        incididunt ut labore et dolore magna aliqua. Rhoncus dolor purus non enim praesent
        elementum facilisis leo vel. Risus at ultrices mi tempus imperdiet. Semper risus in
      </Typography>
    </Grid>
  );
};

ChartContainer.propTypes = {
  classes: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    errorMessage: state.messages.error,
    successMessage: state.messages.success,
    messageConfirmed: state.messages.confirmed
  };
}

export default connect(mapStateToProps, {})(withStyles(Container)(ChartContainer));
