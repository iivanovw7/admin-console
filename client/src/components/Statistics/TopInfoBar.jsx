import { List } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { Buttons } from '../UI/ThemeProperties';
import { TopInfoBarElement } from './TopInfoBarElement';

export const TopInfoBar = props => {

  const { viewBranch, viewMode, viewGroup } = props;

  return (
    <List style={{ paddingTop: 0, paddingBottom: 0, flexDirection: 'row', display: 'flex' }}>
      <TopInfoBarElement title={'View'} icon={'view_module'} view={viewMode} {...props}/>
      {viewGroup && (
        <TopInfoBarElement title={'Group'} icon={'business_center'} view={viewGroup} {...props}/>
      )}
      {viewBranch && (
        <TopInfoBarElement title={'Branch'} icon={'location_city'} view={viewBranch} {...props}/>
      )}
    </List>
  );
};

TopInfoBar.propTypes = {
  classes: PropTypes.object.isRequired,
  viewBranch: PropTypes.string,
  viewGroup: PropTypes.string,
  viewMode: PropTypes.string.isRequired
};

export default withStyles(Buttons, { withTheme: true })(TopInfoBar);