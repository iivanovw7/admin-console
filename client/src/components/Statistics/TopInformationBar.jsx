import { Avatar, Icon, List, ListItem, ListItemText } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { Buttons } from '../UI/ThemeProperties';

export const TopInformationBar = props => {

  const { classes, viewBranch, viewMode, viewGroup } = props;

  return (
    <List style={{ paddingTop: 0, paddingBottom: 0 }}>
      <div className={classes.mobile} style={{ display: 'flex' }}>
        <h4>{viewMode}</h4>
      </div>
      <ListItem className={classes.desktop}>
        <Avatar>
          <Icon>view_module</Icon>
        </Avatar>
        <ListItemText primary="View Mode" secondary={viewMode}/>
      </ListItem>
      {viewGroup && (
        <ListItem className={classes.desktop}>
          <Avatar>
            <Icon>timeline</Icon>
          </Avatar>
          <ListItemText primary="Branch view" secondary={viewGroup}/>
        </ListItem>
      )}
      {viewGroup && (
        <ListItem className={classes.desktop}>
          <Avatar>
            <Icon>timeline</Icon>
          </Avatar>
          <ListItemText primary="Group view" secondary={viewBranch}/>
        </ListItem>
      )}
    </List>
  );
};

TopInformationBar.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(Buttons, { withTheme: true })(TopInformationBar);