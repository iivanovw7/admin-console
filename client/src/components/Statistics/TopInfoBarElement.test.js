import { withStyles } from '@material-ui/core';
import { createMount } from '@material-ui/core/test-utils';
import React from 'react';
import { TopInfoBarElement } from '../../components/Statistics/TopInfoBarElement';
import { NavigationStyles } from '../../components/UI/ThemeProperties';

let wrapper;
let mount;
const Composition = withStyles(NavigationStyles, { withTheme: true })(TopInfoBarElement);
const props = {
  icon: 'test',
  view: 'ADMIN',
  title: 'TITLE'
};

beforeEach(() => {
  mount = createMount();
});

afterEach(() => {
  mount.cleanUp();
});

describe('Should render charts top bar element ', () => {

  it('Should render component correctly field ', () => {
    wrapper = mount(<Composition {...props} />);
    expect(wrapper.find('li').length).toEqual(1);
  });

  it('Should render component with correct data ', () => {
    wrapper = mount(<Composition {...props} />);
    expect(wrapper.find('span').first().text()).toBe('test');
    expect(wrapper.find('span').last().text()).toBe('TITLE');
    expect(wrapper.find('p').text()).toBe('ADMIN');
  });

  it('Should match snapshot', () => {
    wrapper = mount(<Composition {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

});
