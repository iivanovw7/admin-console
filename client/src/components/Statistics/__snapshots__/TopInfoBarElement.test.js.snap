// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render charts top bar element  Should match snapshot 1`] = `"<li class=\\"MuiListItem-root-175 MuiListItem-default-178 MuiListItem-gutters-183\\"><div class=\\"MuiAvatar-root-187 MuiAvatar-colorDefault-188\\"><span class=\\"material-icons MuiIcon-root-190\\" aria-hidden=\\"true\\">test</span></div><div class=\\"MuiListItemText-root-199\\"><span class=\\"MuiTypography-root-205 MuiTypography-subheading-212 MuiListItemText-primary-202\\">TITLE</span><p class=\\"MuiTypography-root-205 MuiTypography-body1-214 MuiTypography-colorTextSecondary-238 MuiListItemText-secondary-203\\">ADMIN</p></div></li>"`;
